18_links_lab/linkslab.txt

```diff
@@ -9,12 +9,33 @@ Preparation: Perform the following  (no need to write the commands you used):
   - Create a symbolic link named file3 that points to "file1"
   Answer: 
 
+ln -s file1 file3
+
   - Create a hard link named file4 that points to "file2"
   Answer: 
 
+ln file2 file4
+
   - Create a new directory named "tmp" in your home folder:
   Answer:
 
+mkdir -p tmp2 /homes/nadavg
+
 2. Move "file3" to the new "~/tmp" directory you just created. Can you still use it to see the original "file1" ? Why not? How would you create a symbolic link that would survive the transfer to ~/tmp ?
 
+mv file3 /homes/nadavg/tmp/
+cat: file3: No such file or directory
+
+I do not know how to create a symbolic link that would survive the transfer to ~/tmp
+Did you mean using $ln without the -s?
+Is it still symbolic?
+
```

Comment:
Removing the -s will create a hard link, so that's not the answer.

The key here is that symbolic link is a file that contains the PATH to another file/directory. that path can be absolute or relative.

A symbolic link to a relative path looks like this:
file3 -> file1
And is created with `ln -s file1 file3`
It means file3 will link to a file named `file1` that is in the same folder as file3. Move file3 and you'll need to also move file1 to get them working.

An absolute link looks like this: `file -> /etc/passwd`, i.e. it starts with a `/`. This means no matter where the link is located, it will always point to the same destination.

You create absolute symbolic links by specifying the full path to their target, such as: `ln -s /etc/passwd file`, or in your case `ln -s /home/nadavg/lab/links/file1 file3`

Since the first argument to ln is an absolute path, the result link will be absolute.

BTW a nice shortcut uses the command realpath to automatically generate the full path to a file, so this line works just the same:
`ln -s $(realpath file1) file3`